core/src/main/java/nl/nn/adapterframework/pipes/MessageSendingPipe.java

```diff
@@ -758,10 +760,18 @@ protected PipeRunResult sendMessage(Message input, PipeLineSession session, ISen
 				}
 			}
 			try {
-				if (sender instanceof IStreamingSender && canStreamToNextPipe() && getOutputValidator()==null && getOutputWrapper()==null && !isStreamResultToServlet()) {
+				if (sender instanceof IStreamingSender && canStreamToNextPipe()) {
 					sendResult =  ((IStreamingSender)sender).sendMessage(input, session, getNextPipe());
+				} else if (sender instanceof IForwardNameProvidingSender) {
```

Comment:
Zou leuk zijn als streaming (capable) senders die niet `canStreamToNextPipe` toch een piperunresult kunnen teruggeven.
Dan is het wel mooi als beide interfaces de zelfde methodenaam delen...

Reply:
Maar de PipeRunResults zijn niet compatibel. Bij streaming krijg je van getForward() een echte forward terug, bij forwardNameProviding een forward waarvan alleen maar de naam geldig is, en die nog geresolved moet worden.